src/query/set_query.rs

```diff
@@ -0,0 +1,201 @@
+use tantivy_fst::raw::CompiledAddr;
+use tantivy_fst::{Automaton, Map};
+
+use crate::query::{AutomatonWeight, Query, Weight};
+use crate::schema::Field;
+use crate::{Searcher, Term};
+
+/// A Term Set Query matches all of the documents containing any of the Term provided
+///
+/// Terms not using the right Field are discared.
+#[derive(Debug, Clone)]
+pub struct TermSetQuery {
+    field: Field,
+    terms: Vec<Term>,
+}
+
+impl TermSetQuery {
+    /// Create a Term Set Query
+    pub fn new<T: IntoIterator<Item = Term>>(field: Field, terms: T) -> Self {
+        let mut terms: Vec<_> = terms
+            .into_iter()
+            .filter(|key| key.field() == field)
+            .collect();
+        terms.sort_unstable();
```

Comment:
Ok for `Iterator -> Vec -> sort -> Dedup`.

Note it is slower  `Iterator -> HashSet -> Vec -> sort`  if the average number of occurence is high, but we can hope users are not feeding us with random stuff here.